include/SFML/System/Angle.hpp

```diff
@@ -0,0 +1,471 @@
+#include <SFML/System/Export.hpp>
+#include <cassert>
+
+
+namespace sf
+{
+////////////////////////////////////////////////////////////
+/// \brief Represents an angle value.
+///
+////////////////////////////////////////////////////////////
+class Angle
+{
+public:
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Default constructor
+    ///
+    /// Sets the angle value to zero.
+    ///
+    ////////////////////////////////////////////////////////////
+    constexpr Angle();
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Return the angle's value in degrees
+    ///
+    /// \return Angle in degrees
+    ///
+    /// \see asRadians
+    ///
+    ////////////////////////////////////////////////////////////
+    [[nodiscard]] constexpr float asDegrees() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Return the angle's value in radians
+    ///
+    /// \return Angle in radians
+    ///
+    /// \see asDegrees
+    ///
+    ////////////////////////////////////////////////////////////
+    [[nodiscard]] constexpr float asRadians() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Return the angle wrapped to the range [-180°, 180°)
+    ///
+    /// \return Angle wrapped to [-180°, 180°)
+    ///
+    /// \see wrappedTo360
+    ///
+    ////////////////////////////////////////////////////////////
+    [[nodiscard]] constexpr Angle wrappedTo180() const;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Return the angle wrapped to the range [0°, 360°)
+    ///
+    /// \return Angle wrapped to [0°, 360°)
+    ///
+    /// \see wrappedTo180
+    ///
+    ////////////////////////////////////////////////////////////
+    [[nodiscard]] constexpr Angle wrappedTo360() const;
+
+    ////////////////////////////////////////////////////////////
+    // Static member data
+    ////////////////////////////////////////////////////////////
+    SFML_SYSTEM_API static const Angle Zero; //!< Predefined 0 degree angle value
+
+private:
+
+    friend constexpr Angle degrees(float);
+    friend constexpr Angle radians(float);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Construct from a number of degrees
+    ///
+    /// This function is internal. To construct angle values,
+    /// use sf::radians or sf::degrees instead.
+    ///
+    /// \param degrees Angle in degrees
+    ///
+    ////////////////////////////////////////////////////////////
+    constexpr explicit Angle(float degrees);
+
+private:
+
+    ////////////////////////////////////////////////////////////
+    // Member data
+    ////////////////////////////////////////////////////////////
+    float m_degrees; //!< Angle value stored as degrees
+};
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
```

Comment:
A small documentation remark: `\relates` is not needed for friend functions.
Removing `\relates` and adding the parameter name `angle` in the friend declaration makes doxygen happy.
Same remark for `degrees` function.

Reply:
Good catch. Removing that `\relates` line actually made the generated docs more expressive. I also added the parameter name to the declaration so it would show up more readily in the docs.